src/eventql/transport/http/api_servlet.cc

```diff
@@ -329,6 +329,122 @@ void APIServlet::fetchTableDefinition(
   res->addBody(buf);
 }
 
+static ReturnCode subSchemaFromJSON(
+    msg::MessageSchema* schema,
+    json::JSONObject::const_iterator begin,
+    json::JSONObject::const_iterator end,
+    size_t* id) {
+  auto ncols = json::arrayLength(begin, end);
+
+  for (size_t i = 0; i < ncols; ++i) {
+    auto col = json::arrayLookup(begin, end, i); // O(N^2) but who cares...
+    *id += 1;
+    auto name = json::objectGetString(col, end, "name");
+    if (name.isEmpty()) {
+      return ReturnCode::error("kRuntimeError", "missing field: name");
+    }
+
+    auto type = json::objectGetString(col, end, "type");
+    if (type.isEmpty()) {
+      return ReturnCode::error("kRuntimeError", "missing field: type");
+    }
+
+    auto type_size = json::objectGetUInt64(col, end, "type_size");
+    auto optional = json::objectGetBool(col, end, "optional");
+    auto repeated = json::objectGetBool(col, end, "repeated");
+
+    auto field_type = msg::fieldTypeFromString(type.get());
+
+    if (field_type == msg::FieldType::OBJECT) {
+      auto child_schema_json = json::objectLookup(col, end, "subcolumns");
+      if (child_schema_json == end) {
+        return ReturnCode::error("kRuntimeError", "missing field: subcolumns");
+      }
+
+      auto p_id = *id;
```

Comment:
this does not make a lot of sense. either we're starting the child ids back from 0/1 or we assign globally unique ids (we should do the latter). however, this is doing neither of both things